codegen-winter/src/air/mod.rs

```diff
@@ -2,34 +2,45 @@ use super::{AirIR, Impl, Scope};
 use ir::TransitionConstraintDegree;
 
 mod boundary_constraints;
-use boundary_constraints::add_fn_get_assertions;
+use boundary_constraints::{add_fn_get_assertions, add_fn_get_aux_assertions};
 
 mod transition_constraints;
-use transition_constraints::add_fn_evaluate_transition;
+use transition_constraints::{add_fn_evaluate_aux_transition, add_fn_evaluate_transition};
+
+mod public_inputs;
+use public_inputs::add_public_inputs_struct;
 
 // HELPERS TO GENERATE AN IMPLEMENTATION OF THE WINTERFELL AIR TRAIT
 // ================================================================================================
 
 /// Updates the provided scope with a new Air struct and Winterfell Air trait implementation
 /// which are equivalent the provided AirIR.
 pub(super) fn add_air(scope: &mut Scope, ir: &AirIR) {
+    // add the Public Inputs struct and its base implementation.
+    add_public_inputs_struct(scope, ir);
+
     let name = ir.air_name();
 
     // add the Air struct and its base implementation.
-    add_air_struct(scope, name);
+    add_air_struct(scope, ir, name);
 
     // add Winterfell Air trait implementation for the provided AirIR.
     add_air_trait(scope, ir, name);
 }
 
 /// Updates the provided scope with a custom Air struct.
-fn add_air_struct(scope: &mut Scope, name: &str) {
+fn add_air_struct(scope: &mut Scope, ir: &AirIR, name: &str) {
     // define the custom Air struct.
-    scope
+    let air_struct = scope
         .new_struct(name)
         .vis("pub")
         .field("context", "AirContext<Felt>");
 
+    // add public inputs
+    for (pub_input, pub_input_size) in ir.public_inputs() {
+        air_struct.field(pub_input, format!("[Felt; {}]", pub_input_size));
+    }
+
```

Comment:
@grjte Since we already have a PublicInputs struct defined, should this just have that as the field or should we not define the PublicInputs struct?

Reply:
I think either way is fine. Let's stick with what you've done for now, where it's directly in the struct, since it's the simplest option. When we revisit public inputs to think about more complex implementations we can think about changing this.